### ISSUE TITLE
ds_pretrain_gpt_125M_MoE64.sh didn't convergence, loss fly after 3k steps?

### ISSUE THREAD

**Issue by jerryli1981** [microsoft/Megatron-DeepSpeed#39]
Hi guys,
In MoE branch, I tested ds_pretrain_gpt_125M_MoE64.sh
training loss is going up after 3k steps?

Is there any suggestions? By the way, which ds version to use for running this branch

**Comment by jerryli1981** [comment]
if lr=4.5e-4, the loss curve will increase at 3k step
if lr=1e-4, the loss cure will increase at 16.25k step

**Comment by jerryli1981** [comment]
hi @shjwudp Thanks for your reply. I am using DeepSpeed v0.6.0. But still not convergence. Could you please share more experiments, which shell do you use to test and how many gpu cards?

Below is my experiments
V100-32G, 8 gpus. the script I use is ds_pretrain_gpt_125M_MoE64.sh. I am quite sure if lr=4e-5 , the loss will fly.

**Comment by hawklucky** [comment]
> hi @shjwudp Thanks for your reply. I am using DeepSpeed v0.6.0. But still not convergence. Could you please share more experiments, which shell do you use to test and how many gpu cards?
> 
> Below is my experiments V100-32G, 8 gpus. the script I use is ds_pretrain_gpt_125M_MoE64.sh. I am quite sure if lr=4e-5 , the loss will fly.

I met the same problem, the grad overflow occurs many time, did you solve it？

**Comment by shjwudp** [comment]
My training curve with ds_pretrain_gpt_125M_MoE64.sh (64 experts will exceed the GPU memory limit, so I used 32 experts instead):  
https://tensorboard.dev/experiment/GNMjjudUQyGehamvhJKlMA/#scalars

I think there are two points worth mentioning:

1. There is a small problem in v0.6.0 that will cause abnormal exit when building expert process group, I fixed it, you can use [this patch](https://github.com/microsoft/DeepSpeed/commit/1e61c7a860e7ac63cd4024a654b4f319063b9374) to do the same fix. 
2. I use [OpenWebText2](https://www.eleuther.ai/projects/owt2/) for the dataset, which is preprocessed in [the officially recommended way](https://github.com/microsoft/Megatron-DeepSpeed#data-preprocessing).

Hope it helps.
